### DISCUSSION TITLE
RTSP Push from Server to another

### DISCUSSION THREAD

**Discussion by 0x524a** [bluenviron/gortsplib#69] [Questions]
Is it possible to push a published stream from one RTSP server to another, without invoking a client ReadFrame from localhost.
In my current Server setup, during OnRecord callback, I am invoking 2 clients,
- one which DialRead on localhost:8554 and ReadFrame
- another client DialPublish to the destination server and WriteFrame

This doesn't seems like an optimal way, but I couldn't figure out anything else.
@aler9 Could you please suggest a better way to handle this?

**Comment by aler9** [comment]
Hello, in my opinion you can avoid the 1st client (not the 2nd one) by hooking into the source server's OnFrame:

```go
func (sh *serverHandler) OnFrame(ctx *gortsplib.ServerHandlerOnFrameCtx) {
    if thisSessionIsPublishing {
        // send frames from this server to the destination server
        clientThatIsPublishing.WriteFrame(ctx.TrackID, ctx.StreamType, ctx.Payload)
    }
}
```
